src/main/java/org/kainos/ea/api/AuthService.java

```diff
@@ -0,0 +1,66 @@
+package org.kainos.ea.api;
+
+import com.password4j.Hash;
+import com.password4j.Password;
+import com.password4j.types.Argon2;
+import org.eclipse.jetty.util.security.Credential;
+import org.kainos.ea.cli.Login;
+import org.kainos.ea.cli.Register;
+import org.kainos.ea.client.FailedToGenerateTokenException;
+import org.kainos.ea.client.FailedToLoginException;
+import org.kainos.ea.client.FailedToRegisterException;
+import org.kainos.ea.db.AuthDao;
+import org.kainos.ea.core.LoginValidator;
+import org.kainos.ea.core.RegisterValidator;
+
+import javax.crypto.SecretKeyFactory;
+import java.security.Security;
+import java.sql.SQLException;
+
+public class AuthService {
+
+    private AuthDao authDao = new AuthDao();
+    private LoginValidator loginValidator = new LoginValidator();
+    private RegisterValidator registerValidator = new RegisterValidator();
```

Comment:
these should be passed in via a constructor so you can mock them out in unit tests